src/main/java/digital/slovensko/autogram/core/Autogram.java

```diff
@@ -62,8 +62,12 @@ public void startVisualization(SigningJob job) {
             } catch (Exception e) {
                 Runnable onContinue = () -> ui.showVisualization(new UnsupportedVisualization(job), this);
 
-                ui.onUIThreadDo(
-                        () -> ui.showIgnorableExceptionDialog(new FailedVisualizationException(e, job, onContinue)));
+                if (UserSettings.load().isCorrectDocumentDisplay()) {
```

Comment:
Neviem ci sa mi uplne paci tento singleton, bude to rozlezene vsade takto. Nevieme to injektnut do `Autogram` a nasledne posielat len co kam treba?